### PR TITLE
Retry string completions from the inferred type by default

### LINKED ISSUE #53475: Autocompletion is working in TS 4.7 to 4.9 but broken/gone in TS 5
# Bug Report

### 🔎 Search Terms

autocompletion

### 🕗 Version & Regression Information

- This changed between versions ^4.7.0 and ^5
- I was unable to test this on prior versions because some type features such as string literal manipulation are not available before 4.7

### ⏯ Playground Link

[Playground link with relevant code](https://tsplay.dev/wgD9lN)

### 💻 Code

Please go to the end of the code and try the `defineConfig` function:

```ts
defineConfig({
  exclude: {
    data: ["<cursor here, ctrl+space>"],
    structure: {
      schemas: ["secret", "<cursor here, ctrl+space>"],
      tables: ["<cursor here, ctrl+space>"]
    }
  }
})
```

### 🙁 Actual behavior

Between TS 4.7 and 4.9, we have autocompletion available for all array values under the `exclude` key
At TS 5 and above (I tried nightly as well), the autocompletion is gone

### 🙂 Expected behavior

The autocompletion should work the same as with TS 4.7 and 4.9

### PR DESCRIPTION
**Pull request by Andarist**
fixes #53475

I'm using the same strategy to retry those requests as the one used in https://github.com/microsoft/TypeScript/pull/52875/files

cc @andrewbranch

### CODE REVIEW

File: src/services/stringCompletions.ts [hunk 1]
```diff
@@ -439,7 +439,7 @@ function getStringLiteralCompletionEntries(sourceFile: SourceFile, node: StringL
             const literals = contextualTypes.types.filter(literal => !tracker.hasValue(literal.value));
             return { kind: StringLiteralCompletionKind.Types, types: literals, isNewIdentifier: false };
         default:
-            return fromContextualType();
+            return fromContextualType() || fromContextualType(ContextFlags.None);
```
Comment (Andarist) [inline src/services/stringCompletions.ts:442]: a more conservative approach would be to do this:
```ts
        case SyntaxKind.ArrayLiteralExpression:
            return fromContextualType() || fromContextualType(ContextFlags.None);
        default:
            return fromContextualType();
```

i went for a broad fix~ but perhaps some node types should not be subjects for those retries?

**Review by andrewbranch — APPROVED** [review]

**Comment by jakebailey** [comment]
Is this something we need to be backporting? Per https://github.com/microsoft/TypeScript/issues/53475#issuecomment-1481973265 this is a 5.0 regression, so it's probably good to get in before we send it out?

**Comment by DanielRosenwasser** [comment]
@typescript-bot user test tsserver
@typescript-bot test tsserver top100
@typescript-bot perf test this

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick this to release-5.0

**Comment by DanielRosenwasser** [comment]
I'll tentatively cherry-pick.

**Comment by Andarist** [comment]
Thanks guys!

**Pull request merged**